common/src/test/java/com/thoughtworks/go/config/materials/mercurial/HgMaterialTest.java

```diff
@@ -692,4 +689,24 @@ void shouldReplaceUrlForCommandLineWithUrlForDisplay_whenCredentialsAreProvidedA
             assertThat(info).isEqualTo(hgMaterial.getUriForDisplay());
         }
     }
+
+    @Test
+    void populateEnvContextShouldSetMaterialEnvVars() {
+        HgMaterial material = new HgMaterial("https://user:password@example.github.com", "folder");
```

Comment:
could we modify this test to contain a path parameter too? `https://user:pass@example.github.com/bob/my-project`?

Reply:
And perhaps explicitly set a non-default branch as well?

Reply:
Refactored to include the suggested changes. Added a separate test to check for default value being set.